sort/merge_sort.go

```diff
@@ -0,0 +1,34 @@
+package sort
+
+func mergeSort(arr []int) []int {
```

Comment:
### Prerequisite Checks

The function `mergeSort` assumes that the input array `arr` is non-nil. However, it's good practice to explicitly check for nil inputs to avoid runtime panics. Consider adding a check at the beginning of the function:

```go
if arr == nil {
    return []int{}
}
```